osu.Game/Scoring/ScoreManager.cs

```diff
@@ -36,10 +38,12 @@ public ScoreManager(RulesetStore rulesets, Func<BeatmapManager> beatmaps, Storag
 
         protected override ScoreInfo CreateModel(ArchiveReader archive)
         {
-            if (archive == null)
+            string filename = archive?.Filenames.FirstOrDefault(f => f.EndsWith(".osr"));
```

Comment:
this logic is changing upstream, and shouldn't be done like this (see https://github.com/ppy/osu/pull/4985).

Reply:
Reverted this back, check is no longer needed